pandas/tests/extension/base/dtype.py

```diff
@@ -112,3 +112,7 @@ def test_construct_from_string_wrong_type_raises(self, dtype):
             match="'construct_from_string' expects a string, got <class 'int'>",
         ):
             type(dtype).construct_from_string(0)
+
+    def test_astype_empty_dataframe(self, dtype):
+        empty_dataframe = pd.DataFrame()
+        assert empty_dataframe.astype(dtype) == empty_dataframe
```

Comment:
can you use tm.assert_frame_equal(result, empty_dataframe) here.

Reply:
done!